cmd/argocd/commands/plugin_test.go

```diff
@@ -0,0 +1,145 @@
+package commands
+
+import (
+	"fmt"
+	"os"
+	"testing"
+)
+
+type testPluginHandler struct {
+	pluginsDirectory string
+	validPrefixes    []string
+
+	// lookup results
+	lookedup  bool
+	lookupErr error
+
+	// execution results
+	executed           bool
+	executedPluginPath string
+	withArgs           []string
+	withEnv            []string
+}
+
+func (t *testPluginHandler) LookForPlugin(filename string) (string, bool) {
+	t.lookedup = true
+
+	dir, err := os.Stat(t.pluginsDirectory)
+	if err != nil {
+		t.lookupErr = err
+		return "", false
+	}
+
+	if !dir.IsDir() {
+		t.lookupErr = fmt.Errorf("expected %q to be a directory", t.pluginsDirectory)
+		return "", false
+	}
+
+	plugins, err := os.ReadDir(t.pluginsDirectory)
+	if err != nil {
+		t.lookupErr = err
+		return "", false
+	}
+
+	filenameWithSuportedPrefix := ""
+	for _, prefix := range t.validPrefixes {
+		for _, p := range plugins {
+			filenameWithSuportedPrefix = fmt.Sprintf("%s-%s", prefix, filename)
+			if p.Name() == filenameWithSuportedPrefix {
+				t.lookupErr = nil
+				return fmt.Sprintf("%s/%s", t.pluginsDirectory, p.Name()), true
+			}
+		}
+	}
+
+	t.lookupErr = fmt.Errorf("unable to find a plugin executable %q", filenameWithSuportedPrefix)
+	return "", false
+}
+
+func (t *testPluginHandler) ExecutePlugin(executablePath string, cmdArgs, environment []string) error {
+	t.executed = true
+	t.executedPluginPath = executablePath
+	t.withArgs = cmdArgs
+	t.withEnv = environment
+	return nil
+}
+
+func Test_ArgoCDPluginHandler(t *testing.T) {
+	cmd := NewCommand()
+	cmd.SilenceErrors = true
+	cmd.SilenceUsage = true
+	tests := []struct {
+		name               string
+		args               []string
+		expectedPluginPath string
+		expectPluginArgs   []string
+		expectLookupError  string
+	}{
+		{
+			name:               "test that normal commands are able to be executed, when no plugin overshadows them",
+			args:               []string{"argocd", "cluster", "list"},
```

Comment:
```suggestion
			args:               []string{"argocd", "cluster", "list"},
```

Is there a way to test the normal argocd command execution for test cases?  The test case won't run remotely since argocd api server is not running. 

cc @leoluz @blakepettersson